cscs-checks/microbenchmarks/latency/latency.py

```diff
@@ -0,0 +1,96 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.16-dev0')
+@rfm.simple_test
+class CPULatencyTest(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.sourcepath = 'latency.cpp'
+        self.build_system = 'SingleSource'
+        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                              'ault:intel', 'ault:amdvega', 'tave:compute']
+        self.valid_prog_environs = ['PrgEnv-gnu', 'PrgEnv-cray']
+        self.num_tasks = 0
+        self.num_tasks_per_node = 1
+
+        self.build_system.cxxflags = ['-O3']
+
+        self.executable_opts = ['16000', '128000', '8000000', '500000000']
+
+        if self.current_system.name in {'daint', 'dom'}:
+            self.modules = ['craype-hugepages1G']
+        if self.current_system.name in {'tave'}:
+            self.modules = ['craype-hugepages512M']
+
+        self.sanity_patterns = sn.assert_eq(
+            sn.count(sn.findall(r'latency', self.stdout)),
+            self.num_tasks_assigned * len(self.executable_opts))
+
+        self.perf_patterns = {
+            'latencyL1': sn.extractall(
+                r'latency \(ns\): (?P<bw>\S+) clocks',
```

Comment:
Since the pattern is common, I wouldn't repeat in all perf patterns here.

Reply:
I'll need to fix this tomorrow morning, the pattern needs to be size specific, otherwise
it doesn't work when run on multiple nodes simultaneously.